src/sonic-bgpcfgd/staticroutebfd/main.py

```diff
@@ -0,0 +1,750 @@
+        else:
+            if_ip = key.split('|')
+            if len(if_ip) < 2:
+                return False, False, "", ""
+            if_name = if_ip[0]
+            value = if_ip[1]
+            valid, is_ipv4, ip = check_ip(value)
+            log_debug("get ip from intf key: valid %s is_ipv4 %s, if_name %s ip %s"%(str(valid), str(is_ipv4), if_name, ip))
+            return valid, is_ipv4, if_name, ip
+
+    def set_local_db(self, table, key, data):
+        try:
+            self.local_db[table][key] = data
+            return
+        except:
+            log_err("set_local_db error, table %s key %s"%(table, key))
+            pass
+
+    def get_local_db(self, table, key):
+        try:
+            v = self.local_db[table][key]
+            return v
+        except:
+            return {}
+
+    def remove_from_local_db(self, table, key):
+        if table in self.local_db:
+            if key in self.local_db[table]:
+                del self.local_db[table][key]
+
+    def append_to_nh_table_entry(self, nh_key, ip_prefix):
+        entry = self.get_local_db(LOCAL_NEXTHOP_TABLE, nh_key)
+        entry.add(ip_prefix)
+
+    def remove_from_nh_table_entry(self, nh_key, ip_prefix):
+        entry = self.get_local_db(LOCAL_NEXTHOP_TABLE, nh_key)
+        if ip_prefix in entry:
+            entry.remove(ip_prefix)
+            if len(entry) == 0:
+                self.remove_from_local_db(LOCAL_NEXTHOP_TABLE, nh_key)
+
+    def set_bfd_session_into_appl_db(self, key, data):
+        fvs = swsscommon.FieldValuePairs(list(data.items()))
+        self.bfd_appl_tbl.set(key, fvs)
+        log_debug("set bfd session to appl_db, key %s, data %s"%(key, str(data)))
+
+    def del_bfd_session_from_appl_db(self, key):
+        self.bfd_appl_tbl.delete(key)
+
+    def interface_set_handler(self, key, data):
+        valid, is_ipv4, if_name, ip = self.get_ip_from_key(key)
+        if not valid:
+            return True
+        value = self.get_local_db(LOCAL_INTERFACE_TABLE, if_name)
+        if len(value) == 0:
+            value = {is_ipv4: ip}
+        else:
+            value[is_ipv4] = ip
+        self.set_local_db(LOCAL_INTERFACE_TABLE, if_name, value)
+        return True
+
+    def interface_del_handler(self, key):
+        valid, is_ipv4, if_name, ip = self.get_ip_from_key(key)
+        if not valid:
+            return True
+        value = self.get_local_db(LOCAL_INTERFACE_TABLE, if_name)
+        if len(value) == 0:
+            return
+        else:
+            value[is_ipv4] = ""  #remove the IP address for the interface
+        self.set_local_db(LOCAL_INTERFACE_TABLE, if_name, value)
+
+    def find_interface_ip(self, if_name, ip_example):
+        valid, is_ipv4, nh = check_ip(ip_example)
+        if not valid:
+            return False, ""
+
+        value = self.get_local_db(LOCAL_INTERFACE_TABLE, if_name)
+        ip = value.get(is_ipv4, "")
+        if len(ip)>0: #ip should be verified when add to local_db
```

Comment:
how do we make sure we process Static Route Config after processing Interface/Portchannel Interface table ?
it seems there is no dependency we have created ? for chassis we don;t want to use Loopback4096 or Loopback0 . if we don;t find ip in interface table can we put setting up that BFD session in hold state (in cache) and can we do retry once it is available ?

Reply:
see the reconciliation function, we process interface table first before process static route table. after that, it will be event driven.

Reply:
updated.